pkg/querier/querier.go

```diff
@@ -50,6 +50,9 @@ type Config struct {
 	// However, we need to use active query tracker, otherwise we cannot limit Max Concurrent queries in the PromQL
 	// engine.
 	ActiveQueryTrackerDir string `yaml:"active_query_tracker_dir"`
+	// LookbackDelta determines the time since the last sample after which a time
+	// series is considered stale.
+	LookbackDelta time.Duration `yaml:"lookback_delta"`
```

Comment:
From a "breaking change" perspective, the flag should remain the same. I will think about how the file can be `promql: -> lookback_delta:`

Reply:
I agree we can't touch the CLI flag (because otherwise would be a breaking change). Have you considered adding a `PromQLConfig` struct in the top level Cortex config and passing `promqlConfig` to the querier as well? @pstibrany do you have smarter ideas?

Reply:
> Have you considered adding a PromQLConfig struct in the top level Cortex config and passing promqlConfig to the querier as well

That's what I was planning to do

Reply:
What other fields would `promqlConfig` have? It feels little empty to have single field in it :-(

I would suggest we deprecate `-promql.lookback-delta` (but keep it for now), and introduce new `-querier.lookback-delta` to match the field, and use `-querier.lookback-delta` if it is set, and `-promql.lookback-delta` otherwise. Can we do that?

Reply:
Actually we can have both `f.DurationVar` using same field. And flag package set it using one or the other flag, based on which one it finds.

Reply:
WDYT?

Reply:
> Actually we can have both f.DurationVar using same field

Does it work? We should make sure it doesn't cause any troubles with defaults. Anyway, if it works fine, LGTM.

(but may not work with the doc generator tool, not sure, remember to check it as well)